db/sqlinterfaces.c

```diff
@@ -3668,8 +3668,14 @@ static int check_sql_access(struct sqlthdstate *thd, struct sqlclntstate *clnt)
     if (gbl_check_access_controls) {
         check_access_controls(thedb);
         gbl_check_access_controls = 0;
+        /* Force all clients to  reauthenticate if passwords have changed. */
+        clnt->authenticated = 0;
```

Comment:
How will this force all clients to re-authenticate?

Reply:
There's an if branch right below this line, checking if clnt->authenticated is true. If not, perform password check.

Reply:
Right, but that only re-auth's this client.

Reply:
Once gbl_check_access_controls is set to true, all clients will invalidate the flag. gbl_check_access_controls  is set to true when 1) authentication has been enabled; 2) a user's password has changed.

Reply:
Yes you're right - only the 1st client who gets in here will re-authenticate.